### ISSUE TITLE
NoScript sometimes runs scripts in wrong order

### ISSUE THREAD

**Issue by dextercd** [hackademix/noscript#151]
Since a few days ago I started experiencing errors while using DuckDuckGo with Firefox with the NoScript plugin. After a bit of experimenting I found that the scripts on DuckDuckGo get run in the wrong order..

Here's the error I'm seeing:

```
Uncaught TypeError: window.locale is undefined
    filter https://duckduckgo.com/g2391.js:1
    canRollDice https://duckduckgo.com/g2391.js:1
    rollDice https://duckduckgo.com/g2391.js:1
    addExperimentParameters https://duckduckgo.com/g2391.js:1
    addExperimentParameters https://duckduckgo.com/g2391.js:1
    nrj https://duckduckgo.com/d2831.js:1
    nrji https://duckduckgo.com/?q=noscript&t=ffab:1
    <anonymous> https://duckduckgo.com/util/u471.js:1
    b4 https://duckduckgo.com/lib/l118.js:26
    fireWith https://duckduckgo.com/lib/l118.js:26
    ready https://duckduckgo.com/lib/l118.js:15
    bX https://duckduckgo.com/lib/l118.js:15
```

It says that window.locale is undefined even though it is initialized in the first \<script\> in \<head\>. With the debugger I confirmed that some \<body\> scripts are sometimes run before the \<head\> script.

I've made a minimal project that reproduces this error in [this repository](https://github.com/dextercd/noscript-problem-minimal-reproducible-case).

It's a simple index.html file with a few script elements. Head contains two script elements, the first one does console.log('head1') and sets window.locale, the second script in head only does console.log('head2'). Body contains a script that does console.log('body') and tries to use window.locale.
Sometimes this runs without any issues, in that case I see head1, head2, body in the correct order in the logs, other times it does this:

![image](https://user-images.githubusercontent.com/14349965/89738310-cb993400-da77-11ea-90a9-cfbe6120e6c3.png)

In this case the order was head2, body,  head1. Since body uses window.locale which is set in head1 it causes an error.

I think it's caused by some kind of race condition/timing issue, random variations to the index.html file causes the problem to go away. That's the reason why there's an additional big but useless JavaScript file and why it references an non-existent css file, without those elements the issue doesn't occur as much. It also seems like disabling caching increases the error rate.

I was able to reproduced this issue on the latest version of NoScript and with the development version.

I've only found one other person experiencing this problem [link](https://addons.mozilla.org/en-US/firefox/addon/noscript/reviews/1579394/).

**Comment by hackademix** [comment]
Please check whether this still happens on [11.0.37rc2](https://noscript.net/getit#devel), thanks.

**Comment by dextercd** [comment]
I only see a download link for version 11.0.37rc1 on that page, but I think the version number just hasn't been updated. I still see this problem when using that version.

I also tried a build from March which also has this problem, so I don't think it's caused by a recent change in NoScript. My guess is that this issue has existed for a while but only manifests now due to a recent change in DuckDuckGo. I also thought it might've been introduced by a bug/change in Firefox, but I just tried an older version of Firefox and see that this issue still occurs.
